2.43

ATSchoolENikonov opened the pull request:
Сделал задание, хорошо бы проверить)

methshest committed:
2.43
Сделал задание, хорошо бы проверить)

Separador commented on Home_2.43/Task_2.java:
```diff
@@ -0,0 +1,32 @@
+                    break;
+                case 7:
+                    System.out.println("Sunday");
+                    break;
```
Этот `break` необязателен.

Separador reviewed (changes requested):
1. В первой задаче при аргументах `1 1 1` печатает "Условия не соблюдены". О каких условиях речь?

2. В четвертой задаче при аргументах `1 1 11` печатает `obtuse`, хотя нельзя построить треугольник с такими сторонами. Также при аргументах `5 4 3` пишет `acute`, а при аргументах `3 4 5` `right`.

3. В пятой задаче:
- при аргументах `1 1 1 1` печатает "Not enough money to buy", хотя очевидно суммы хватает
- при аргументах `1 1 1 2` печатает "U'r payback: 1.0000000000000002", что считает что-то не то.  Плюс в задании указано вывести отдельно число рублей и копеек.

Также в целом рекомендую тексты ошибки, особенно когда ты знаешь в чем ошибка, писать недвусмысленно и подробнее. Потом самому легче разбираться.

Также можно чуть упростить логику программы для понимания, если уменьшить число вложенных конструкций. Если будет время, то попробуй, а так править необязательно. Например, для второй задачи :
```
if (x < 1 || x > 7) {
    System.out.println("There is mistake");
    return;
}
// далее мы знаем, что с x все ок
switch (x) {
...
}
```
или в третьей
```
if (a <= 0 || a > 30000) {
    System.out.println("There is mistake");
    return;
}

if ((a % 4 == 0 && a % 100 != 0) || (a % 400 == 0)) {
    System.out.println("YES");
} else {
    System.out.println("NO");
}
```

methshest committed:
2.43
Поправил, должно работать

Separador reviewed (changes requested):
1) Почему не подходит вариант, когда какие-то числа равны?
Например, если числа '1 1 1', наибольшим будет 1.
Если мы не попали в первое условие, то о у нас выбор либо b или c. Поправь второе условие на (b > c). Если в него не попадем останется только c.

2) В четвертом задании на вводах `1 1 1` и `3 3 2` выводит `right`, а должно acute. При вычислении макс стороны и остальные подходит условие из первой задачи

3) В четвертой задаче при вводе `0 1 3 0` пишет сдачу 3 рубля и -1 копейка.
Приведи цену и платеж к копейкам и с ними выполняй вычисления.
А при выводе уже раздели на рубли и копейки.

methshest committed:
2.43
Поправил условие и я так полагаю в целом я намудрил, такое условие для трех параметров универсально)
В пятой задаче вычисления сделал красиво как мне кажется, посомтри плиз или есть косяк, вроде хорошо вычисляет)
